src/main/java/org/jenkinsci/plugins/workflow/multibranch/WorkflowMultiBranchProjectFactory.java

```diff
@@ -30,16 +30,28 @@
 import jenkins.scm.api.SCMSource;
 import jenkins.scm.api.SCMSourceCriteria;
 import org.kohsuke.stapler.DataBoundConstructor;
+import org.kohsuke.stapler.DataBoundSetter;
+import java.io.IOException;
 
 /**
  * Defines organization folders by {@link WorkflowBranchProjectFactory}.
  */
 public class WorkflowMultiBranchProjectFactory extends AbstractWorkflowMultiBranchProjectFactory {
+    private WorkflowBranchProjectFactory workflowBranchProjectFactory;
 
-    @DataBoundConstructor public WorkflowMultiBranchProjectFactory() {}
+    @DataBoundSetter
+    public void setScriptPath(String scriptPath) {
+        workflowBranchProjectFactory.setScriptPath(scriptPath);
+    }
+
+    public String getScriptPath() { return workflowBranchProjectFactory.getScriptPath(); }
+
+    @DataBoundConstructor public WorkflowMultiBranchProjectFactory() {
+        workflowBranchProjectFactory = new WorkflowBranchProjectFactory();
+    }
 
     @Override protected SCMSourceCriteria getSCMSourceCriteria(SCMSource source) {
-        return new WorkflowBranchProjectFactory().getSCMSourceCriteria(source);
+        return workflowBranchProjectFactory.getSCMSourceCriteria(source);
```

Comment:
And here you would create a new `WorkflowBranchProjectFactory` with that `scriptPath`, as you had before…